runtime/stdlib.js

```diff
@@ -352,85 +352,129 @@ function caml_compare_val (a, b, total) {
   var stack = [];
```

Comment:
It looks like this stack is only pushed onto.  Is that correct?

Reply:
No, we pop from it as well.